File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dict = {}
    for c in completion:
        if c not in dict:
            dict[c] = 1
        else:
            dict[c] += 1
    for p in participant:
        if p in dict:
            dict[p] -= 1
        else:
            answer = p
    for k, v in dict.items():
        if v < 0:
            answer = k
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participants, String[] completion) {
        String answer = "";
        Map<String, Integer> name = new HashMap<>();
        for (String participant : participants) {
            name.put(participant, name.getOrDefault(participant, 0) + 1);
        }
        for (String comple : completion) {
            name.put(comple, name.getOrDefault(comple, 0) + 1);
        }
        
        for (Map.Entry<String, Integer> entry : name.entrySet()) {
            String key = entry.getKey();
            Integer value = entry.getValue();
            if (value % 2 == 1) {
                return key;
            }
        }
        return answer;
    }
}